python/ray/tune/ray_trial_executor.py

```diff
@@ -180,14 +180,14 @@ def _stop_trial(self, trial, error=False, error_msg=None,
         try:
             trial.write_error_log(error_msg)
             if hasattr(trial, "runner") and trial.runner:
+                trial.runner.stop.remote()
                 if (not error and self._reuse_actors
                         and self._cached_actor is None):
                     logger.debug("Reusing actor for {}".format(trial.runner))
                     self._cached_actor = trial.runner
                 else:
                     logger.debug(
                         "Destroying actor for trial {}.".format(trial))
-                    trial.runner.stop.remote()
```

Comment:
@ericl This makes it so that `trial_executor.stop_trial(t)` actually calls `_stop` on the trainable, even in actor reuse.

Reply:
Uh, that defeats the point of actor reuse here. Why is this an issue, doesn't actor reuse not work anyways unless reset_config is properly implemented?

Reply:
The issue I was trying to address was here: https://github.com/ray-project/ray/issues/5767

The underlying complication is that with `reuse_actors`, `_stop` is now the same as a `cleanup` + `on_training_end` hook; where you only want to skip the  `cleanup` hook but still run `on_training_end` (where you do things like what the PR is doing)